anvil/observables.py

```diff
@@ -153,16 +148,24 @@ def bootstrap_function(func, states, *args, n_boot=100):
         dimension
 
     """
-    boot_index = torch.randint(0, states.shape[0], size=(states.shape[0], n_boot))
-    # put boot index on final dimension
-    resampled_states = states[boot_index, :].transpose(1, 2)
-    res = func(resampled_states, *args)
-    return res
+    # NOTE: Optimal seems to be around n_sample x batch_size = 10000 for
+    # calc_two_point_function on my laptop. Clearly this is not a general
+    # rule though.
+    sample_size = states.shape[0]
+    batch_size = min(n_boot, max(1, 10000 // sample_size))
+    n_batch = n_boot // batch_size
```

Comment:
do we actually get what we asked for with //, isn't that floor divide?

Reply:
Ok, I've switched to using `ceil` so we get `n_boot` resamples at minimum. I don't think we need to be more precise than this?

Reply:
I should have been more explicit in asking what you think about this. If you think it's immaterial and n_boot is just approximate then I guess that's ok. I just think it's neat if when I bootstrap with n_boot = 1000 then I get 1000 resamples